### ISSUE TITLE
Last Song in Playlist Not Finishing Playback

### ISSUE THREAD

**Issue by MochaMage** [daneren2005/Subsonic#1052]
I've noticed that when I play an album or playlist that is not pinned for offline listening, the last song in the playlist will consistently fail to finish playing back completely. The track will partially play but eventually stop. The track list makes it seem as if buffering the track is failing, with the percentage constantly increasing and decreasing. I've confirmed that this issue is not specific to my Airsonic server as I am able to play entire albums using Substreamer so it seems to be specific to Dsub.

**Comment by Bomfunk** [comment]
I am experiencing this too. Sometimes the last song in the playlist won't even download to play. DSub seems to get stuck on it until I try to stream some other song.

**Comment by MochaMage** [comment]
> I am experiencing this too. Sometimes the last song in the playlist won't even download to play. DSub seems to get stuck on it until I try to stream some other song.

Yup, that happens to me as well, you and I seem to have the exact same issue. Most of the time, part of the song plays but then I get times where the last song just won't play at all.

**Comment by gravelld** [comment]
I've seen reports of this as well. What devices and OS versions are you using?

**Comment by MochaMage** [comment]
> I've seen reports of this as well. What devices and OS versions are you using?

I'm using a Google Pixel 3a, Lineage OS 12 and on the server side, Ampache 5 (Haven't tested with 6 yet)
